test/methods.jl

```diff
@@ -802,6 +802,47 @@ end
     end
 end
 
+@testset "Base.stack" begin
+    a = [1 2 3; 4 5 6]
+    da = DimArray(a, (X(4.0:5.0), Y(6.0:8.0)))
+    b = [7 8 9; 10 11 12]
+    ca = DimArray(b, (X(4.0:5.0), Y(6.0:8.0)))
+    db = DimArray(b, (X(6.0:7.0), Y(6.0:8.0)))
+
+    x = DimArray([da, db], (Z(4.0:5.0)))
+    @test_warn "Lookup values" stack(x)
+
+    x = DimArray(fill(da, 2, 2), (Dim{:a}(4.0:5.0), Dim{:b}(6.0:7.0)))
+    sx = stack(x)
+    @test sx isa AbstractDimArray
+    @test dims(sx) == (dims(first(x))..., dims(x)...)
+
+    @test_throws "ArgumentError" stack(x; dims=4)
+
+    x = DimArray([da, ca], (Dim{:a}(1:2),))
+    sx = stack(x; dims=1)
+    sy = @test_nowarn stack(x; dims=:)
+    sz = @test_nowarn stack(x; dims=X)
```

Comment:
What happens with `dims=Z` ?

Reply:
Currently, `dims=Z` (where Z is not a dimension in the DimArrays being stacked) falls back to adding the new dimension at the end (so the result is the same as for the default behavior):
```julia
_maybe_dimnum(x, dim) = hasdim(x, dim) ? dimnum(x, dim) : ndims(x) + 1
```

Should we throw an error instead, like when an out-of-range Integer dim is give? I.e.:
```julia
function _maybe_dimnum(x, dim::Int)
    if dim < ndims(x) + 2
        return dim
    else
        throw(ArgumentError(LazyString("cannot stack slices ndims(x) = ", ndims(x) + 1, " along dims = ", dim)))
    end
end
```

Reply:
I was thinking we should add a dimension like it is currently, but we can set it to be a `Z` dimension

Reply:
Using the current minimal approach, stacking with a new dimension `Z` would be:
```julia
a = [1 2 3; 4 5 6]
da = DimArray(a, (X(4.0:5.0), Y(6.0:8.0)))
b = [7 8 9; 10 11 12]
db = DimArray(b, (X(4.0:5.0), Y(6.0:8.0)))
x = DimArray([da, db], (Dim{:a}(1:2),))

stack(set(x, :a=>Z)) # Or
set(stack(x), :a=>Z})
```

We could have this behavior be automatic with:
```julia
function Base.stack(iter::AbstractArray{<:AbstractDimArray}; dims=:)
    x = Base._stack(_maybe_dimnum(first(iter), dims), check_stack_dims(iter))
    if !hasdim(x, dims) && Z isa Union{Dimension,Type{<:Dimension}}
        x = set(x, DimensionalData.dims(x)[end] => dims)
    end
    return x
end
```
But I wonder if it could be confusing to have this different behavior for when `dims <: Dimension` and `hasdim(x, dims)` (the new dimension is inserted before `dims` but keeps the original name) versus `!hasdims(x, dims)` (the new dimension is inserted at the end and renamed to `dims`). It also means the return type can't be inferred

Reply:
> falls back to adding the new dimension at the end (so the result is the same as for the default behavior)

I'm not sure we are understanding each other... My only ask here was that this new dimension is a `Z` dimension if `dims=Z` was used